In [1]:
import numpy as np
import pandas as pd

####  Задание-1

In [ ]:
# Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
# zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
# ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
# Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
# Полученные значения должны быть равны.
# Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
# а затем с использованием функций из библиотек numpy и pandas.

In [ ]:
# Решение: воспользуемся формулой для рассчета ковариации:
# cov = M(XY) - M(X)*M(Y)

In [32]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [40]:
cov = np.mean(zp*ks) - (np.mean(zp) * np.mean(ks))
cov

9157.839999999997

In [41]:
# Проверим результат с помощью функции:
np.cov(zp,ks,ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [ ]:
# Коэффициент корелляции Пирсона наъходим по формуле5
# r(XY) = COV(XY) / (sig(X)*sig(Y))

In [42]:
r = cov / (np.std(zp) * np.std(ks))
r

0.8874900920739158

In [39]:
# Проверим результат с помощью функции:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

*Ответ: ковариация двух этих величин равна 9157.84, а коэффициент корелляции Пирсона 0.88749009.*

#### Задание-2

In [ ]:
# Измерены значения IQ выборки студентов,
# обучающихся в местных технических вузах:
# 131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
# Известно, что в генеральной совокупности IQ распределен нормально.
# Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [ ]:
# Решение: так как сигма нам не известна, то используем t-таблицу, 
# а учитывая, что выборка состоит из 10 измерений - для рассчета сигмы 
# применяем формулу для несмещенного значения.
# Интервал находим по формуле X(+/-)t(a/2)*sig/sqrt(n)

In [10]:
set = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [11]:
X = set.mean()
X

118.1

In [14]:
# Значение t определяем по таблице:
t = 2.262
sig = set.std(ddof=1)
sig

10.54566788359614

In [15]:
n = len(set)
n

10

In [16]:
int1 = X - t * (sig / np.sqrt(n))
int1

110.55660776308164

In [17]:
int2 = X + t * (sig / np.sqrt(n))
int2

125.64339223691834

*Ответ: доверительный интервал находится в границах [110.56; 125.64].*

#### Задание-3

In [ ]:
# Известно, что рост футболистов в сборной распределен нормально
# с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
# среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
# ожидания с надежностью 0.95.

In [ ]:
# Решение: так как сигма нам известна, то используем Z-таблицу, 
# Интервал находим по формуле X(+/-)Z(a/2)*sig/sqrt(n)

In [23]:
D = 25
n = 27
Z = 1.96 
X = 174.2

In [24]:
sig = np.sqrt(D)
sig

5.0

In [27]:
int1 = X - Z * (sig / np.sqrt(n))
int1

172.31398912064722

In [28]:
int2 = X + Z * (sig / np.sqrt(n))
int2

176.08601087935276

*Ответ: доверительный интервал для математического ожидания находится в гшраницах [172.31; 176.09].*